include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,661 @@
+ *
+ * @return Number of input packets consumed (0 ... num_pkt)
+ * @retval <0 on failure
+ */
+int odp_comp_op(const odp_packet_t pkt_in[],
+		      const odp_packet_t pkt_out[],
+		      odp_comp_packet_op_param_t param[],
+		      odp_comp_packet_result_t result[],
+		      int num_pkt);
+
+/**
+ * Asynchronous packet comp
+ *
+ * This operation does packet compression/decompression in asynchronous mode.
+ * A successful submitted entry means it has been queued for processing but
+ * not fully processed. Each submitted entry results will be notified via
+ * event after operation on that entry is complete. On received event w.r.t
+ * each submitted packet, application should call odp_comp_result() to retrieve
+ * operational result.
+ * For every input packet entry in in_pkt array, user should pass
+ * corresponding valid output packet handle.On return, API returns with
+ * number of entries successfully submitted for operation.
+ * In case of partially accepted array i.e. when
+ * number of packets returned < num_pkt, application my attempt to resubmit
+ * subsequent entries via calling any of the operation API.
+ *
+ * User should compute processed data len = output_data_range.len - digest_len,
+ * where digest_len queried through odp_comp_hash_alg_capability().
+ * Application should monitor ODP_EVENT_PACKET
+ * be queued with subtype ODP_EVENT_PACKET_COMP on queue.
+ *
+ * for compression + hash, call returns with hash appended at the end of
+ * packet on which odp_packet_op_param_t:last field is set.
+ *
+ * If any of the packet has status code set to ODP_COMP_ERR_OUT_OF_SPACE in
+ * its result, application should call operation API again with corresponding
+ *  output packet available with free space.
+ *
+ * Please note it is always recommended that application using async mode,
+ * provide sufficiently large buffer size to avoid ODP_COMP_ERR_OUT_OF_SPACE.
+ * Else it is recommended that application set required user context on packet
+ * before calling each odp_comp_op_enq() to correctly retrieve required
+ * information associated with packet.
+ *
+ * @param         pkt_in   Input Packets
+ * @param         pkt_out  Output Packet to store output of corresponding entry
+ * @param         param    Operation parameters array
+ * @param         num_pkt  Number of packets to be processed
+ *
+ * @return Number of input packets consumed (0 ... num_pkt)
+ * @retval <0 on failure
+ */
+int odp_comp_op_enq(const odp_packet_t pkt_in[],
+			  const odp_packet_t pkt_out[],
+			  const odp_comp_packet_op_param_t param[],
+			  int num_pkt);
+
+/**
+ * Get Comp operation results from processed packet.
+ *
+ * Successful Comp operation on async type produce packet which contain
+ * comp result metadata. This function copies operation results from an Comp
+ * processed packet. Event subtype of this kind of packet is
+ * ODP_EVENT_PACKET_COMP. Results are undefined if non-Comp processed packet is
+ * passed as input.
```

Comment:
Actually, shouldn't this be called for every `ODP_EVENT_PACKET` of subtype `ODP_EVENT_PACKET_COMP`? You don't know what the actual result is (success or failure) until you get its `odp_comp_packet_result_t` from this call.